admiral/pkg/clusters/virtualservice_routing.go

```diff
@@ -619,3 +620,115 @@ func getMeshHTTPPortForRollout(ports map[string]map[string]uint32) (uint32, erro
 func getMeshHTTPPortForDeployment(ports map[string]map[string]uint32) (uint32, error) {
 	return getMeshHTTPPort(common.Deployment, ports)
 }
+
+// getDRHostsFromRouteDestinations generates the DR hosts from the route destinations
+// These will be a map of .svc.cluster.local hosts which will be later
+// used to create the DestinationRule for VS based routing
+func getDRHostsFromRouteDestinations(destinations map[string][]*networkingV1Alpha3.RouteDestination) (map[string]string, error) {
+
+	if destinations == nil {
+		return nil, fmt.Errorf("destinations map is nil")
+	}
+
+	drHosts := make(map[string]string)
+	for _, routeDestinations := range destinations {
+		for _, routeDestination := range routeDestinations {
+			host := routeDestination.Destination.Host
+			if host == "" {
+				return nil, fmt.Errorf("host is empty")
+			}
+			drHosts[host] = host
+		}
+	}
+
+	return drHosts, nil
+}
+
+// addUpdateDestinationRuleForSourceIngress adds or updates the DestinationRule for the source ingress
+// This is where the DestinationRules are created for the cross-cluster VS based routing
+// The DestinationRule is created for the .svc.cluster.local hosts that were discovered during the discovery phase
+// on each source cluster
+func addUpdateDestinationRuleForSourceIngress(
+	ctx context.Context,
+	ctxLogger *log.Entry,
+	remoteRegistry *RemoteRegistry,
+	sourceClusterToDRHosts map[string]map[string]string,
+	sourceIdentity string) error {
+
+	if sourceIdentity == "" {
+		return fmt.Errorf("sourceIdentity is empty")
+	}
+
+	san := fmt.Sprintf("%s%s/%s", common.SpiffePrefix, common.GetSANPrefix(), sourceIdentity)
+
+	for sourceCluster, drHosts := range sourceClusterToDRHosts {
```

Comment:
Question would it better to create one DR with host like - `"*.sandbox-sandbox-vinayclient2-usw2-qal.svc.cluster.local`

Reply:
I dont see any reason why we cant do that. ok, let me make this change